lib/graphql/schema/wrapper.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module GraphQL
+  class Schema
+    class Wrapper
+      include GraphQL::Schema::Member::CachedGraphQLDefinition
+      include GraphQL::Schema::Member::TypeSystemHelpers
+
+      attr_reader :of_type
+      
+      def initialize(of_type)
+        @of_type = of_type
+      end
+
+      def to_graphql
+        raise NotImplementedError
+      end
+
+      def unwrap
+        @of_type.unwrap
+      end
+
+      def ==(other)
+        self.class == other.class && of_type == other.of_type
```

Comment:
🤔 does this need to account for intermediate wrappers? i.e. should `[Foo] == `[Foo!]`?

Reply:
Oh wait, you're comparing the `of_type`s directly not using the recursion in `unwrap`, so... the recursion happens through the repeated `==` calls instead and does the right thing. 🤦‍♂️

Reply:
Yep exactly!